plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -51,42 +53,47 @@ func parseDeploymentFromReplicaSet(name string) string {
 		return ""
 	}
 
-	if !stringInRuneset(suffix, kubeAllowedStringAlphaNums) {
+	if containUnexpectedRuneSet(suffix, deploymentUnexpectedRegEx) {
 		// Invalid suffix
 		return ""
 	}
 
 	return name[:lastDash]
 }
 
-// get the cronJob name by stripping the last dash following some rules
-// return empty if it is not following the rule
+// Get the cronJob name by stripping the last dash following by the naming convention: JobName-UnixTime
+// based on https://github.com/kubernetes/kubernetes/blob/c4d752765b3bbac2237bf87cf0b1c2e307844666/pkg/controller/cronjob/cronjob_controllerv2.go#L594-L596.
+// Before v1.21 CronJob in Kubernetes has used Unix Time in second; after v1.21 is a Unix Time in Minutes.
+
 func parseCronJobFromJob(name string) string {
 	lastDash := strings.LastIndexAny(name, "-")
+
+	//Return empty since the naming convention is: JobName-UnixTime, if it does not have the "-", meanings the job name is empty
 	if lastDash == -1 {
-		// No dash
 		return ""
 	}
+
 	suffix := name[lastDash+1:]
-	if len(suffix) != 10 {
-		// Invalid suffix if it is not 10 rune
+	suffixInt, _ := strconv.ParseInt(suffix, 10, 64)
+	//Convert Unix Time In Minutes to Unix Time
+	suffixStringMultiply := strconv.FormatInt(suffixInt*60, 10)
```

Comment:
Do we want to always do this check? I thought this was a breaking change in only some clusters?

Reply:
Yes we always want to. Since AWS [has supported version 1.21 ](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html). Moreover, I haven't able to found a way in checking whether the CronJob is a CronJob or a ControllerV2. Therefore, leading to this solution though.